orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db.sh

```diff
@@ -54,69 +42,50 @@ errorExit() {
   exit 1
 }
 
-dump_to_azure() {
+kinit_as_hdfs() {
+  # Use a new Kerberos credential cache, to keep from clobbering the default.
+  export KRB5CCNAME=/tmp/krb5cc_cloudbreak_$EUID
+  HDFS_KEYTAB=/run/cloudera-scm-agent/process/$(ls -t /run/cloudera-scm-agent/process/ | grep hdfs-NAMENODE$ | head -n 1)/hdfs.keytab
+  doLog "kinit as hdfs using Keytab: $HDFS_KEYTAB"
+  kinit -kt "$HDFS_KEYTAB" "hdfs/$(hostname -f)"
+  if [[ $? -ne 0 ]]; then
+    errorExit "Couldn't kinit as hdfs"
+  fi
+}
+
+move_backup_to_cloud () {
+  LOCAL_BACKUP="$1"
+  kinit_as_hdfs
+  doLog "INFO Uploading to ${BACKUP_LOCATION}"
+
+  hdfs dfs -mkdir -p "$BACKUP_LOCATION"
```

Comment:
here are were are creating a directory and accessing directly. S3 provides an eventual guarantee. 
In short, if you create a directory and try to write to it immediately, it might fail.

we use S3 guard to handle this issue with S3. S3 guard tracks all the operations and changes the metadata.

I'm not sure if S3 guard is enabled by default. Is there anything that has to be done for it. Can you check with @risdenk  on this?

Reply:
So there are potentially a few options here:
* retry on failure to upload
* Upload the whole folder instead of trying to create it then put files in it
* in theory s3guard is enabled so this isn't even something that needs to be worried about

Reply:
I'm going to be lazy and rely on S3 guard to protect me from the eventual consistency problems.